### PR TITLE
Modifier key platform names

### PR DESCRIPTION
**Pull request by siikamiika**
Resolves https://github.com/FooSoft/yomichan/issues/504.

Things related to environment are extracted from backend into the `Environment` class because otherwise there would be a circular dependency between `Backend` and `profile-conditions.js`. The dependency issue could be fixed, but the modifier key definitions would probably still be too much bloat for `Backend` so they can be kept in a dedicated class.

I went with key symbols for macOS and names elsewhere, because the usage of the symbols seems so widespread on mac. Some examples (sourced from https://www.nngroup.com/articles/ui-copy/):

There also seems to be an `X` button in the Evernote screenshot, but I didn't add that for any OS yet.

I think it's fine to use `''` as a separator on macOS for now, but this could change in the future if a key name consists of multiple characters.

I'm not sure if https://github.com/FooSoft/yomichan/commit/de6135251c56fb3f2247540c3c4e1a229e228be2 could be done with datasets and CSS, but I went that way because it was simple. `settingsPopulateModifierKeys` could be in the wrong place.

### CODE REVIEW

File: ext/bg/js/profile-conditions.js [hunk 1]
```diff
@@ -36,135 +39,140 @@ function _profileConditionTestDomainList(url, domainList) {
     return false;
 }
 
-const _profileModifierKeys = [
-    {optionValue: 'alt',   name: 'Alt'},
-    {optionValue: 'ctrl',  name: 'Ctrl'},
-    {optionValue: 'shift', name: 'Shift'}
-];
+let profileConditionsDescriptor = null;
```
Comment (toasted-nutbread) [inline ext/bg/js/profile-conditions.js:42]: At some point, this file should be refactored into a class. This isn't necessarily due to changes in this commit; there are several scripts which store state and/or have private global functions that should be converted to classes at some point.

This is basically just a heads up as the file grows in complexity.

File: ext/mixed/js/environment.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            return 'firefox';
+        } else {
+            return 'chrome';
+        }
+    }
+
+    _getModifierInfo(browser, os) {
+        let osModifierKeys;
+        let osModifierSeparator;
+        if (os === 'win') {
```
[the remark this answers is not in the document]
Comment (siikamiika) [reply ext/mixed/js/environment.js]: https://github.com/FooSoft/yomichan/pull/519/commits/4070da1c685a90b6336658b74b72daf0cadb970f

File: ext/mixed/js/environment.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                // NOP
+            }
+            return 'firefox';
+        } else {
+            return 'chrome';
+        }
+    }
+
+    _getModifierInfo(browser, os) {
+        let osModifierKeys;
```
[the remark this answers is not in the document]
Comment (siikamiika) [reply ext/mixed/js/environment.js]: https://github.com/FooSoft/yomichan/pull/519/commits/fcae697c946542e9f6517709364fd1a9bafd801f

`keys` is actually built using the raw definitions so I kept the `os` prefix on the raw ones but now the return object uses the short syntax.

File: ext/bg/js/profile-conditions.js (hunk 1 above)
Comment (siikamiika) [reply ext/bg/js/profile-conditions.js:42]: Yeah, I was thinking the same. The environment dependency could be given to the constructor by either `Backend` or settings.

**Pull request merged**